### PR TITLE
Handle snippets when linting

### PR DESCRIPTION
**Pull request by TAGraves**
We need to prevent you from trying to lint snippets, and we don't want to automatically lint snippets if you don't specify a lint target. Snippets will be incidentally linted as long as they are used.

### CODE REVIEW

File: internal/cli/service_test.go [hunk 1]
```diff
@@ -1557,5 +1557,46 @@ Checked 2 files and found 4 problems.
...
+				Expect(memfs.WriteFileString(".mint/_snippet1.yml", ".mint/_snippet1.yml contents")).NotTo(HaveOccurred())
+				Expect(memfs.WriteFileString(".mint/_snippet2.yml", ".mint/_snippet2.yml contents")).NotTo(HaveOccurred())
+
+				lintConfig.OutputFormat = cli.LintOutputOneLine
+			})
+
+			Context("without targeting", func () {
+				It("doesn't target the snippets", func() {
+
+					// lintConfig.MintFilePaths = []string{".mint/base1.yml"}
```
Comment (doxavore) [inline internal/cli/service_test.go:1575]: Should this be removed?

File: internal/cli/service.go [hunk 2]
```diff
@@ -733,6 +740,21 @@ func removeDuplicateStrings(list []string) []string {
 	return slices.Compact(list)
 }
 
+func findSnippets(fileNames []string) ([]string, []string) {
```
Comment (doxavore) [inline internal/cli/service.go:743]: Minor nit: this is a rare case where I'd probably use named return values to make it a little more clear what this is returning.

**Review by doxavore — APPROVED** [review]

File: internal/cli/service_test.go (hunk 1 above)
Comment (TAGraves) [reply internal/cli/service_test.go:1575]: ah yeah, thanks. good catch!

File: internal/cli/service.go (hunk 2 above)
Comment (TAGraves) [reply internal/cli/service.go]: done

File: internal/cli/service_test.go (hunk 1 above)
Comment (TAGraves) [reply internal/cli/service_test.go]: fixed

**Review by doxavore — APPROVED** [review]

File: internal/cli/service.go [hunk 3]
```diff
@@ -237,8 +238,14 @@ func (s Service) Lint(cfg LintConfig) (*api.LintResult, error) {
 			}
 		}
 		targetPaths = removeDuplicateStrings(targetPaths)
+		_, snippetFileNames := findSnippets(targetPaths)
+		if len(snippetFileNames) > 0 {
+			return nil, fmt.Errorf("You cannot target snippets for linting, but you targeted the following snippets: %s\n\nTo lint snippets, include them from a Mint run definition and lint the run definition.", strings.Join(snippetFileNames, ", "))
```
Comment (dan-manges) [inline internal/cli/service.go:243]: This error message 😍

**Pull request merged**